database/timestamp.go

```diff
@@ -0,0 +1,45 @@
+package database
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"time"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"gorm.io/gorm/schema"
+)
+
+var ErrUnsupportedType = errors.New("unsupported type")
+
+// TimestampSerializer is a GORM serializer that allows the serialization and deserialization of the
+// google.protobuf.Timestamp protobuf message type.
+type TimestampSerializer struct{}
```

Comment:
It would be great if you could add a comment here and maybe in other places about the advantages of using `time.Time` as the internal database representation.

Reply:
I noticed that the PR comment explains this but won't hurt to have it here as well.